File: problems/substrings/substrings.js
let substrings = (s) => {
    let sb = []
    for (let i = 1; i <= s.length; i++) 
        for (let j = 0; j <= s.length - i; j++)
            sb.push(s.substring(j, j+i));
    return sb;
}

substrings("abcd");

File: problems/substrings/substrings.rb
#!/usr/bin/env ruby

def sub_strings(str)
  sub_strs = str.split('')

  str.size.times do |i|
    for j in (i+1...str.size)
      sub_strs << str[i..j]
    end
  end

  sub_strs
end

puts sub_strings('abc').inspect